File: server/todolistJava01/src/main/java/jp/co/linnefromice/todolist01/service/UserService.java
package jp.co.linnefromice.todolist01.service;

import jp.co.linnefromice.todolist01.model.User;
import jp.co.linnefromice.todolist01.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    @Autowired
    private UserRepository repository;

    public Optional<User> findOne(final String id) {
        return repository.findById(id);
    }

    public List<User> findAll() {
        return repository.findAll();
    }

    public User save(User user) {
        return repository.save(user);
    }
}

File: server/todolistKotlin01/src/main/kotlin/jp/co/linnefromice/todolistKotlin01/service/UserService.kt
package jp.co.linnefromice.todolistKotlin01.service

import jp.co.linnefromice.todolistKotlin01.model.Task
import jp.co.linnefromice.todolistKotlin01.model.User
import jp.co.linnefromice.todolistKotlin01.repository.TaskRepository
import jp.co.linnefromice.todolistKotlin01.repository.UserRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserService (
    private val userRepository: UserRepository){

    fun findOne(id: String): Optional<User> {
        return userRepository.findById(id);
    }

    fun findAll(): List<User> {
        return userRepository.findAll();
    }

    fun save(user: User): User {
        return userRepository.save(user);
    }
}